src/client/opamClientConfig.mli

```diff
@@ -27,21 +27,61 @@ type 'a options_fun =
   ?pin_kind_auto:bool ->
   ?autoremove:bool ->
   ?editor:string ->
-  unit -> 'a
+  'a
+  (* constraint 'a = 'b -> 'c *)
 
-val default : t
-
-val set : t -> t options_fun
-
-val setk : (t -> 'a) -> t -> 'a options_fun
-
-val r : t ref
-
-val update : ?noop:_ -> unit options_fun
-
-(** Sets the options, reading the environment to get default
-    values when unspecified *)
-val init: ?noop:_ -> unit options_fun
+include OpamStd.Config.Sig
+  with type t := t
+   and type 'a options_fun := 'a options_fun
 
 (** Extra files included in [opam search] *)
 val search_files: string list
+
+(** Load the global configuration file (opamroot/config) and initialise all opam
+    sub-libraries, overriding the given arguments *)
+
+val opam_init:
+  ?root_dir:OpamTypes.dirname ->
+  ?strict:bool ->
+  ?skip_version_checks:bool ->
+  ?all_parens:bool ->
+  ?log_dir:OpamTypes.dirname ->
+  ?print_stats:bool ->
+  ?sync_archives:bool ->
+  ?pin_kind_auto:bool ->
+  ?autoremove:bool ->
+  ?editor:string ->
+  ?current_switch:OpamSwitch.t ->
+  ?switch_from:[ `Command_line | `Default | `Env ] ->
```

Comment:
The origin we got the "current switch setting" from: it's used in opam to display the proper `eval $(opam config env --switch)` messages for example.
Probably doesn't deserve to be here, at least not in this form -- passing `~switch` should probably set it to `Command_line` automatically.